velox/exec/Window.cpp

```diff
@@ -187,6 +194,46 @@ void Window::createWindowFunctions() {
   }
 }
 
+// Support 'rank', 'dense_rank' and
+// 'row_number' functions and the agg window function with default frame.
+bool Window::supportRowsStreaming() {
+  bool supportsStreaming = false;
+
+  for (const auto& windowNodeFunction : windowNode_->windowFunctions()) {
+    const auto& functionName = windowNodeFunction.functionCall->name();
+    auto windowFunctionMetadata =
+        exec::getWindowFunctionMetadata(functionName).value();
+
+    if (windowFunctionMetadata.processedUnit == ProcessedUnit::kRows) {
+      const auto& frame = windowNodeFunction.frame;
+      bool isDefaultFrame =
+          (frame.startType ==
+               core::WindowNode::BoundType::kUnboundedPreceding &&
```

Comment:
Maybe consider to add a method in window function which takes frame as input to indicate whether support row-wise streaming or not?